pkg/agent/clean/orphan_bindings_test.go

```diff
@@ -0,0 +1,570 @@
+			Labels: map[string]string{
+				"bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb-8k9i4": auth.PrtbInClusterBindingOwner,
+			},
+		},
+		Subjects: groupSubjects,
+	}
+	nonOrphanBindingWithStrangeLabel := v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "non-orphan binding with a new-style PRTB label that has a long namespace name",
+			Labels: map[string]string{
+				"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-37a42": auth.PrtbInClusterBindingOwner,
+			},
+		},
+		Subjects: groupSubjects,
+	}
+	nonOrphanBindingWithSimpleLabel := v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "non-orphan binding with hashed simple label",
+			Labels: map[string]string{
+				"default_foo": auth.PrtbInClusterBindingOwner,
+			},
+		},
+		Subjects: groupSubjects,
+	}
+	nonOrphanBindingWithHashedLabel := v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "non-orphan binding with hashed label",
+			Labels: map[string]string{
+				"p-kmc2v_grouplonglonglonglonglonglonglonglonglonglonglong-a59f5": auth.PrtbInClusterBindingOwner,
+			},
+		},
+		Subjects: groupSubjects,
+	}
+	nonOrphanBindingWithUIDLabel := v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "non-orphan binding with UID label",
+			Labels: map[string]string{
+				"c8bcebc8-779d-4fe7-a367-aa80897fc7b1": auth.PrtbInClusterBindingOwner,
+			},
+		},
+		Subjects: groupSubjects,
+	}
+	nonOrphanUserBinding := v1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "non-orphan user binding",
+			Labels: map[string]string{
+				"ns1-non-existent-prtb": auth.PrtbInClusterBindingOwner,
+			},
+		},
+		Subjects: []v1.Subject{
+			{
+				Kind: v1.UserKind,
+				Name: "Bob",
+			},
+		},
+	}
+	allBindings := []v1.RoleBinding{
+		orphanBindingWithBadOldLabel,
+		orphanBindingWithBadNewLabel,
+		orphanBindingWithStrangeLabel,
+		nonOrphanBindingWithStrangeLabel,
+		nonOrphanBindingWithSimpleLabel,
+		nonOrphanBindingWithHashedLabel,
+		nonOrphanBindingWithUIDLabel,
+		nonOrphanUserBinding,
+	}
+	client := orphanBindingsCleanup{
+		prtbs: &fakePRTBClient{bindings: []v3.ProjectRoleTemplateBinding{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "default",
+					UID:       "c8bcebc8-779d-4fe7-a367-aa80897fc7b1",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "grouplonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong",
+					Namespace: "p-kmc2v",
+					UID:       "abc123",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "whale",
+					Namespace: strings.Repeat("a", 63),
+				},
+			},
+		}},
+		prtbHashes: make(map[string]struct{}),
+		prtbUIDs:   make(map[string]struct{}),
+		roleBindings: &fakeRoleBindingClient{
+			bindings: allBindings,
+		},
+	}
+
+	tests := []struct {
+		name         string
+		dryRun       bool
+		remainingRBs []v1.RoleBinding
+		errExpected  bool
```

Comment:
nit - if you are splitting out the error cases into a separate test function and none of the cases in this table expect an error, just don't include this field in the test struct.